### ISSUE TITLE
Dynamically generated Distill Blog title name

### ISSUE THREAD

**Issue by dcaud** [rstudio/distill#240]
How do you make a dynamically generated Distill Blog title name show correctly in the index/collections of the website?

I am trying to create Distill post titles that are generated from data. The post title shows correctly when the .Rmd file is knitted, but it shows as "untitled" on the index page where I have `listing: posts`.

Here's an example of a post rmd:

```
---
output:
  distill::distill_article:
    self_contained: false
---

```
{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
db <- readRDS("../../in/data.RDS")
focal.row <- data[1,]
```
---
title: `r focal.row$name`
---
```

**Comment by jjallaire** [comment]
I don't think this is currently possible b/c we read the title directly from the YAML. You could write a script that post-processes your site HTML to fixup titles as required.

**Comment by dcaud** [comment]
Thanks for the idea! I was also considering an obnoxious use of cat() to
create the markdown files.

**Comment by apreshill** [comment]
Not tested, but wondering if using parameters might be the way to go here:

https://itsalocke.com/blog/r-quick-tip-parameter-re-use-within-rmarkdown-yaml/

https://bookdown.org/yihui/rmarkdown-cookbook/dynamic-yaml.html (the second solution)
